File: common/src/main/java/dev/biserman/wingscontracts/core/block/state/properties/ContractPortalMode.java
package dev.biserman.wingscontracts.core.block.state.properties;
import net.minecraft.util.StringRepresentable;

public enum ContractPortalMode implements StringRepresentable {
    UNLIT("UNLIT", 0, "unlit"), 
    LIT("LIT", 1, "lit"),
    COIN("COIN", 2, "coin");
    
    private final String name;

    private ContractPortalMode(final String key, final int i, final String name) {
        this.name = name;
    }
    
    @Override
    public String toString() {
        return this.name;
    }

    public String getSerializedName() {
        return this.name;
    }
}

File: common/src/main/kotlin/dev/biserman/wingscontracts/block/state/properties/ContractPortalMode.kt
package dev.biserman.wingscontracts.block.state.properties

import net.minecraft.util.StringRepresentable

enum class ContractPortalMode(key: String, i: Int) : StringRepresentable {
    UNLIT("UNLIT", 0),
    LIT("LIT", 1),
    COIN("COIN", 2),
    NOT_CONNECTED("NOT_CONNECTED", 3),
    ERROR("ERROR", 4);

    override fun toString(): String {
        return this.name.lowercase()
    }

    override fun getSerializedName(): String {
        return this.name.lowercase()
    }
}
